cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -560,6 +560,87 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
+                            //
+                            // A' = A
+                            // A'' = A
+                            // A = Al * Bl
+                            // A' = A' -> 32
+                            // A' = Ah' * Bl
+                            // A' = A' <- 32
+                            // A = A + A'
+                            // B = B -> 32
+                            // B = Bh * Al''
+                            // B = B <- 32
+                            // A = A + B
+
+                            // A' = A
+                            // A'' = A
+                            let lhs = input_to_reg(ctx, inputs[1]);
+                            let lhs_tmp1 = ctx.alloc_tmp(RegClass::V128, types::I64X2);
+                            let lhs_tmp2 = ctx.alloc_tmp(RegClass::V128, types::I64X2);
+                            ctx.emit(Inst::gen_move(lhs_tmp1, lhs, ty));
+                            ctx.emit(Inst::gen_move(lhs_tmp2, lhs, ty));
+
+                            // A = Al * Bl
+                            let rhs = input_to_reg_mem(ctx, inputs[0]);
+                            let dst = output_to_reg(ctx, outputs[0]);
+                            ctx.emit(Inst::gen_move(dst, lhs, ty));
+                            ctx.emit(Inst::xmm_rm_r(SseOpcode::Pmuludq, rhs.clone(), dst));
+
+                            // A' = A' -> 32
+                            // A' = Ah' * Bl
+                            // A' = A' <- 32
+                            // A = A + A'
+                            ctx.emit(Inst::xmm_rmi_reg(
+                                SseOpcode::Psrlq,
+                                RegMemImm::imm(32),
+                                lhs_tmp1,
+                            ));
+                            ctx.emit(Inst::xmm_rm_r(SseOpcode::Pmuludq, rhs.clone(), lhs_tmp1));
+                            ctx.emit(Inst::xmm_rmi_reg(
+                                SseOpcode::Psllq,
+                                RegMemImm::imm(32),
+                                lhs_tmp1,
+                            ));
+                            ctx.emit(Inst::xmm_rm_r(
+                                SseOpcode::Paddq,
+                                RegMem::reg(lhs_tmp1.to_reg()),
+                                dst,
+                            ));
+
+                            // B = B -> 32
+                            // B = Bh * Al''
+                            // B = B <- 32
+                            // A = A + B
+                            ctx.emit(Inst::xmm_rmi_reg(
+                                SseOpcode::Psrlq,
+                                RegMemImm::imm(32),
+                                Writable::from_reg(rhs.to_reg().unwrap()),
+                            ));
+                            ctx.emit(Inst::xmm_rm_r(
+                                SseOpcode::Pmuludq,
+                                RegMem::reg(lhs_tmp2.to_reg()),
```

Comment:
This is the only use of `lhs_tmp2`, and at this point we could still reuse `lhs`, so we should be able to avoid the `lhs_tmp2` temporary at this point.